### PR TITLE
Database Model and Validations

### PR DESCRIPTION
**Pull request by kensayo**
Working with ActiveRecords on Rails - **Micro Reddit database**

- Created a User table with columns `username` `email` and `password`
- Created a Post table with columns `title` and `link`
- Created a Comment table with columns `comment`
- Relation `user has many post`
- Relation `post has many comments`
- Relation `user has many comments`

### CODE REVIEW

**Review by rindrajosia — CHANGES_REQUESTED** [review]
Invalid :red_circle: 
Please use this file as linter config [.github/workflows](https://github.com/microverseinc/linters-config/blob/master/ror/.github/workflows/linters.yml)
That's it, you can submit another code review request after making these changes :recycle: :innocent:
With all my respect :pray:
Rindra Josia

**Review by taiwo2 — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ⚠️ 🚧
@kensayo Weldone for this project,Nice readme , however for this project to be approved few changes is rquired of you
Highlights:
- [x] A user can be created **(covered by automated tests)** ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/building_with_active_record/creatingUser.png) from [Odin Project](https://www.theodinproject.com/courses/ruby-on-rails/lessons/building-with-active-record-ruby-on-rails))
- [x] A post can be created **(covered by automated tests)** ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/building_with_active_record/creatingPost.png) from [Odin Project](https://www.theodinproject.com/courses/ruby-on-rails/lessons/building-with-active-record-ruby-on-rails))
- [x] A comment can be created **(covered by automated tests)** ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/building_with_active_record/creatingComment.png) from [Odin Project](https://www.theodinproject.com/courses/ruby-on-rails/lessons/building-with-active-record-ruby-on-rails))
- [x] kindly add `rails db:create` to your setup section in readme file
##### Post

- [x] validations are present (`body` and `title` in automated tests but can be anything else) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/building_with_active_record/postValidation.png) from [Odin Project](https://www.theodinproject.com/courses/ruby-on-rails/lessons/building-with-active-record-ruby-on-rails)).should be `text`  for your title

##### Comment

- [x] validations are present (`body` in automated tests but can be anything else) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/building_with_active_record/commentValidation.png) from [Odin Project](https://www.theodinproject.com/courses/ruby-on-rails/lessons/building-with-active-record-ruby-on-rails)). should be `text`
kindly follow project specification.
## once specific changes made kindly request for another review

**Comment by mke2111** [comment]
## Project Under review. Please be patient. ⚪

**Review by mke2111 — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED 🔴 ♻️ 

## Hey Happy Coder, 👋🏽 
**This is a great project you have going on, well-done lad and thanks for implementing the changes requested earlier on**. 

### Well-done 💯 
    1. New users and posts can be created successfully. 👍🏽 
    2. The linter checks run just fine.

**However, there are a few sections to improve before your project is approved.** 🔴 
<hr>

### REQUIRED

- [x] For some reason, I'm not able to create a new comment via the console. Look into this and give it a quick fix.

### OPTIONAL

- [x] Remove all commented-out pieces of code in all your model files.

#

### Kindly make all the requested changes and submit another CRR. 💯 

**Cheers 🍹 and Happy coding 💻** 

<hr>

Mention @mke2111 in this thread and I'll be glad to help. 😄 
### Best regards
Slack 🆘  Roy Mukuye
Twitter 🐦  [Roy Mukuye](https://twitter.com/Roymkenya)
LinkedIn 🔗  [Roy Mukuye](https://www.linkedin.com/in/roy-mukuye/)

File: README.md [hunk 1]
```diff
@@ -1,24 +1,105 @@
...
+
+
+### 👨🏻‍🔧 Setup
+
+- Open your terminal and move to the root of the project, using ```cd``` command.
+- Then you must run ```$ rails db:migrate```, so you can work locally with the database.
+- Run ```$ rails db:create``` so you create the database
+- You must run ```$ bundle install```  
+- Start server using ```$ rails s```
+- Finally go to the Ruby console with `$ rails c`
```
Comment (rindrajosia) [inline README.md]: You should ask the user to use the command **rails bundle install** first then **rails db:create** after that **rails db:migrate** and finally **rails s** and/or **rails c**

**Review by rindrajosia — CHANGES_REQUESTED** [review]
Changes Required :red_circle:
To Highlight: :angel:

- A post can be created :+1: 
- A comment can be created :100: 

To improve :factory_worker: [REQUIRED]:

- [x] Please check the validation related to the **pwd** because I am not able to create a User even if I use the example in your readme.


Nice to have:

- [x]  please check your readme, especially the Setup section. (check my comment below :arrow_double_down: )

That's it, you can submit another code review request after making these changes :recycle: :innocent:
With all my respect :pray:
Rindra Josia

**Review by elisha2kyakpo1 — APPROVED** [review]
# Status: Approved✔🎉💪
**Congratulations! it's time to celebrate!** I'm happy to see all the changes requested being implemented.

Cheers💪🎉

Elisha

**Pull request merged**